src/aggregator/aggregator/aggregator.go

```diff
@@ -368,7 +367,11 @@ func (agg *aggregator) Close() error {
 	if agg.state != aggregatorOpen {
 		return errAggregatorNotOpenOrClosed
 	}
+
 	close(agg.doneCh)
+	// Waiting for the ticking goroutines to return.
+	agg.wg.Wait()
```

Comment:
This seems to fall in-between `agg.Lock()` and `defer agg.Unlock()`. Maybe it would be safer to unlock while waiting to avoid any potential deadlocks?

Reply:
yeah, placement manager `placementTick` will not exit and hang around forever if it tries to RLock() during Close

Reply:
you probably want to lock first to close channel and change state, then unlock, wait and lock again (from a cursory look)

Reply:
You both are right, thanks. Guess I should learn how to code. Updated the PR by restructuring the close flow.